### PR TITLE
feat: from org team invite

### PR DESCRIPTION
**Pull request by sean-brydon**
Loom:  https://www.loom.com/share/b4877aa4f234493bb417874e1d58dcff

CleanShot 2023-06-16 at 13 46 42@2x

How to test:
* Add a bunch of members to ORG
* Open team invite modal 
* Goto Org tab if not already
* profit

### CODE REVIEW

File: packages/features/ee/teams/components/MemberInvitationModal.tsx [hunk 1]
```diff
@@ -259,6 +262,33 @@ export default function MemberInvitationModal(props: MemberInvitationModalProps)
...
+                  required: t("enter_email_or_username"),
+                }}
+                render={({ field: { onChange, value } }) => (
+                  <>
+                    <TeamInviteFromOrg
+                      selectedEmails={value}
+                      handleOnChecked={(userEmail) => {
+                        // If 'value' is not an array, create a new array with 'userEmail' to allow future updates to the array.
+                        // If 'value' is an array, update the array by either adding or removing 'userEmail'.
+                        const newValue = Array.isArray(value)
```
Comment (emrysal) [inline packages/features/ee/teams/components/MemberInvitationModal.tsx:279]:
```suggestion
                        const emailArray = Array.isArray(value) ? value : [value];
                        const newValue = emailArray.includes(userEmail) ? emailArray.filter(email => email !== userEmail) : [...emailArray, userEmail];
```

Better even is to extract it;

```javascript
function toggleElementInArray(value: string[] | string, element: string): string[] {
  const array = Array.isArray(value) ? value : [value];
  return array.includes(element) ? array.filter(item => item !== element) : [...array, element];
}
```

File: packages/features/ee/organizations/components/TeamInviteFromOrg.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+}: TeamInviteFromOrgProps) {
+  const [searchQuery, setSearchQuery] = useState("");
+
+  const filteredMembers = orgMembers?.filter((member) => {
+    if (!searchQuery) {
+      return true; // return all members if searchQuery is empty
+    }
+    const { user } = member ?? {}; // destructuring with default value in case member is undefined
+    return keysToCheck.some((key) => user?.[key]?.toLowerCase().includes(searchQuery.toLowerCase()));
+  });
```
Comment (sean-brydon) [inline packages/features/ee/organizations/components/TeamInviteFromOrg.tsx:29]: Generic filter function used on keysToCheck  -> Filter by name email and username

File: packages/features/ee/organizations/components/TeamInviteFromOrg.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+  console.log({ filteredMembers, orgMembers });
```
Comment (sean-brydon) [inline packages/features/ee/organizations/components/TeamInviteFromOrg.tsx:31]:
```suggestion
```

File: packages/features/ee/teams/components/MemberInvitationModal.tsx [hunk 4]
```diff
@@ -293,20 +324,22 @@ export default function MemberInvitationModal(props: MemberInvitationModalProps)
...
+                    props.token
+                      ? copyInviteLinkToClipboard(props.token)
+                      : createInviteMutation.mutate({ teamId: props.teamId })
+                  }
+                  className={classNames("gap-2", props.token && "opacity-50")}
+                  data-testid="copy-invite-link-button">
+                  <Link className="text-default h-4 w-4" aria-hidden="true" />
+                  {t("copy_invite_link")}
+                </Button>
+              )}
```
Comment (sean-brydon) [inline packages/features/ee/teams/components/MemberInvitationModal.tsx:342]: Invite links dont really work in context of bulk joins/orgs so for now we dont display it unless the option is on individual

File: packages/features/ee/teams/pages/team-members-view.tsx [hunk 5]
```diff
@@ -68,13 +68,25 @@ const MembersView = () => {
...
   const showDialog = router.query.inviteModal === "true";
   const [showMemberInvitationModal, setShowMemberInvitationModal] = useState(showDialog);
   const [showInviteLinkSettingsModal, setInviteLinkSettingsModal] = useState(false);
 
-  const { data: team, isLoading } = trpc.viewer.teams.get.useQuery(
+  const { data: orgMembersNotInThisTeam, isLoading: isOrgListLoading } =
+    trpc.viewer.organizations.getMembers.useQuery(
+      {
+        teamIdToExclude: teamId,
+        distinctUser: true,
```
Comment (sean-brydon) [inline packages/features/ee/teams/pages/team-members-view.tsx:82]: We pass in distinct here as we dont want multiple of the same user ID to be pulled back

**Review by Udit-takkar — COMMENTED** [review]
i am not able run this branch locally 

Screenshot 2023-06-20 at 4 46 50 PM

File: packages/features/ee/teams/pages/team-members-view.tsx (hunk 5 above)
Comment (Udit-takkar) [reply packages/features/ee/teams/pages/team-members-view.tsx:82]: Screenshot 2023-06-20 at 5 18 58 PM

i keep getting too many 404 in this query

**Review by Udit-takkar — COMMENTED** [review]
Even though i had two members in the organization but not in the team. i did not see the organization tab in the invite modal

Screenshot 2023-06-20 at 5 24 40 PM

**Comment by PeerRich** [comment]
hell yeah this looks dope!!

**Review by leog — COMMENTED** [review]
@sean-brydon Tested locally, for some reason I don't see pending org members in the team invite dialog. Maybe I'm missing something? 

I have these members invited to the org:
<kbd></kbd>

And in the invite dialog for Engineering subteam can't see them:
<kbd></kbd>

**Comment by hariombalhara** [comment]
I invited 3 users. 2 with org email and 1 without org email but I don't see non-org email under Organizations tab in invite dialog. Is that expected?
image
image

**Comment by hariombalhara** [comment]
Invited users snapshot from from user table
image

Because non org emails don't have organizationId set they aren't retrieved by getMembers handler.

We need to be very clear on how org members are fetched, it should be fetched in one way consistently.
An org member can be fetched just by seeing if there is a membership with an organization or it can be fetched by checking if user.organizationId is set.

**Comment by hariombalhara** [comment]
There is a critical UI issue that happens when "three dot" dialog is opened and closed. After that I can't take any action on the screen. Can't search. Can't click Add button.

I have seen a similar issue earlier on some other route, a wrongly configured modal is responsible for such a behaviour.

**Comment by hariombalhara** [comment]
UI suggestion:
In this screen we should rename the CTA to "Add user" as we are just adding an existing member of org.
image

**Review by leog — DISMISSED** [review]
Works great, thanks for taking care of this, very essential for orgs.

**Comment by hariombalhara** [comment]
2 issues in this loom. The first one is blocker.
https://www.loom.com/share/10a61444647f46608dd653593cafe618

**Review by hariombalhara — DISMISSED** [review]
Requesting changes because of following important bugs. Will merge once these are resolved.
- https://github.com/calcom/cal.com/pull/9569#issuecomment-1623559769
- https://github.com/calcom/cal.com/pull/9569#issuecomment-1624625470

**Comment by hariombalhara** [comment]
Possibly unrelated issue but an important one: https://github.com/calcom/cal.com/issues/9979

File: packages/features/ee/teams/components/MemberInvitationModal.tsx [hunk 6]
```diff
@@ -59,8 +59,8 @@ interface FileEvent<T = Element> extends FormEvent<T> {
   target: EventTarget & T;
 }
 
-function toggleElementInArray(value: string[] | string, element: string): string[] {
-  const array = Array.isArray(value) ? value : [value];
+function toggleElementInArray(value: string[] | string | undefined, element: string): string[] {
+  const array = value ? (Array.isArray(value) ? value : [value]) : [];
```
Comment (hariombalhara) [inline packages/features/ee/teams/components/MemberInvitationModal.tsx:63]: [Sometimes error comes when adding an existing user to an org ](https://www.loom.com/share/227f966d5541413588ee589b0e460de7)

cc @sean-brydon I don't know why it has been passed an undefined value, TypeScript isn't complaining. But this changes fixes the issue.

**Comment by hariombalhara** [comment]
Another [issue found](https://github.com/calcom/cal.com/issues/9980) while testing this PR. Pretty sure it's unrelated.

**Comment by sean-brydon** [comment]
> There is a critical UI issue that happens when "three dot" dialog is opened and closed. After that I can't take any action on the screen. Can't search. Can't click Add button. https://www.loom.com/share/c3c352df099e47cebb1bc01d74d920cc
> 
> I have seen a similar issue earlier on some other route, a wrongly configured modal is responsible for such a behaviour.

Fixed in latest commit - was a bug in radix-ui they were not removing the pointer-event none on all dialog closures (fixed in latest version)

**Comment by sean-brydon** [comment]
@hariombalhara can't replicate the full page crash - added loom to your issue

**Review by ericrommel — APPROVED** [review]
LGTM but I found a non-blocking issue which can be addressed as a follow-up PR.

https://github.com/calcom/cal.com/issues/9998

**Pull request merged**